test-src/epfl/test7-analysis/TestFusion.scala

```diff
@@ -49,15 +49,14 @@ trait ScalaGenFatArrayLoopsFusionOpt extends ScalaGenArrayLoopsFat with ScalaGen
 // trait NestLambdaProg extends Arith with Functions with Print 
 // --> from TestCodeMotion.scala
 
+@virtualize // for +
 trait FusionProg extends Arith with ArrayLoops with Print {
   
-  implicit def bla(x: Rep[Int]): Rep[Double] = x.asInstanceOf[Rep[Double]]
-  
   def test(x: Rep[Unit]) = {
     
     val constant = array(100) { i => 1 }
 
-    val linear = array(100) { i => 2*i }
+    val linear = array(100) { i => 2*i.toDouble } //NOTE(trans): previously, we could just use 2*i
```

Comment:
This actually seems like a potentially significant problem. For the functions in EmbeddedControls, we don't get the benefit of _any_ explicit conversions now. There's always a more general form of the function in scope for anything that isn't defined explicitly...